<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>SingletCode 1.2 documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/sphinx_rtd_size.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/sphinx_highlight.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="#" class="icon icon-home">
            SingletCode
          </a>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
<li class="toctree-l1"><a class="reference internal" href="index.html#document-mainIntro">Singlet Code</a></li>
<li class="toctree-l1"><a class="reference internal" href="index.html#document-classifier">Classifier</a></li>
<li class="toctree-l1"><a class="reference internal" href="index.html#document-doubletDetectionMethods">Doublet detection methods</a></li>
<li class="toctree-l1"><a class="reference internal" href="index.html#document-datasetsInfo">Datasets used in Singlet Code</a></li>
<li class="toctree-l1"><a class="reference internal" href="index.html#document-references">References</a></li>
<li class="toctree-l1"><a class="reference internal" href="index.html#document-FAQs">FAQs</a></li>
<li class="toctree-l1"><a class="reference external" href="https://www.biorxiv.org/content/10.1101/2023.08.04.552078v2">Link to Paper</a></li>
<li class="toctree-l1"><a class="reference external" href="https://www.goyallab.org/">Lab Website</a></li>
<li class="toctree-l1"><a class="reference external" href="https://github.com/madelinemelzer/singletCode">Github repository</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="#">SingletCode</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="#" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">SingletCode 1.2 documentation</li>
      <li class="wy-breadcrumbs-aside">
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="welcome-to-the-documentation-of-singlet-code">
<h1>Welcome to the documentation of Singlet Code!<a class="headerlink" href="#welcome-to-the-documentation-of-singlet-code" title="Permalink to this heading">¶</a></h1>
<p>Singlet code <a class="reference external" href="https://www.biorxiv.org/content/10.1101/2023.08.04.552078v2">(link)</a> uses cell barcoding technology to identify <strong>real singlets</strong> in scRNAseq data, benchmarks existing doublet detection methods and provides a <em>classifier</em> to detect singlets in <em>unbarcoded</em> datasets as well of the same cell type.</p>
<div class="toctree-wrapper compound">
<span id="document-mainIntro"></span><section id="singlet-code">
<h2>Singlet Code<a class="headerlink" href="#singlet-code" title="Permalink to this heading">¶</a></h2>
<p>This section contains information about the need for doublet detection in RNAseq analysis, singlet code detects singlets, how is this data used for benchmarking other doublet detection methods and the command line interface to</p>
<p>TO-DO:
Add information about command-line interface and figures.</p>
<div class="toctree-wrapper compound">
<span id="document-mainSingletCode/doubletInfo"></span><section id="doublets-in-rna-sequencing-data">
<h3>Doublets in RNA sequencing data<a class="headerlink" href="#doublets-in-rna-sequencing-data" title="Permalink to this heading">¶</a></h3>
<div class="line-block">
<div class="line"><strong>How scRNA-seq technologies work?</strong></div>
<div class="line">scRNA-seq technologies rely on distributing individual cells from a suspension into individual reactions, each labeled with a unique “ID”, usually in the form of a reaction-specific sequence barcode.</div>
<div class="line"><br /></div>
<div class="line"><strong>Since each cell has an unique ID, why are there doublets?</strong></div>
<div class="line">Despite numerous technological optimizations, multiple cells can occasionally be encapsulated in a single reaction, resulting in doublets or multiplets where two or more cells are assigned the same reaction ID. The percentage of doublets in a given experiment depends on several factors, including the features of the sample and throughput, and can be as high as 40-50% <span id="id1">[<a class="reference internal" href="index.html#id3" title="Nicholas J. Bernstein, Nicole L. Fong, Irene Lam, Margaret A. Roy, David G. Hendrickson, and David R. Kelley. Solo: doublet identification in single-cell rna-seq via semi-supervised deep learning. Cell Systems, 11(1):95-101.e5, July 2020. doi:10.1016/j.cels.2020.05.010.">B1</a>]</span> <span id="id2">[<a class="reference internal" href="index.html#id4" title="Nan Miles Xi and Jingyi Jessica Li. Benchmarking computational doublet-detection methods for single-cell rna sequencing data. Cell Systems, 12(2):176-194.e6, February 2021. doi:10.1016/j.cels.2020.11.008.">B3</a>]</span>. In turn, such artifacts affect the downstream analysis <span id="id3">[<a class="reference internal" href="index.html#id5" title="Malte D. Luecken and Fabian J. Theis. Current best practices in single-cell rna-seq analysis: a tutorial. Molecular Systems Biology, 15(6):e8746, June 2019. doi:10.15252/msb.20188746.">B2</a>]</span>.</div>
</div>
</section>
<span id="document-mainSingletCode/working"></span><section id="how-does-singlet-code-detect-singlets">
<h3>How does Singlet Code detect singlets?<a class="headerlink" href="#how-does-singlet-code-detect-singlets" title="Permalink to this heading">¶</a></h3>
<p>Singlet Code uses the fact that the lineage barcodes are within the cell before oil encapsulation for sequencing. Therefore, a cell would have one lineage barcode or a result of multiplicity, more than barcode but with restrictions.
To be labelled a singlet, a barcoded cell needs to satisfy one of these three conditions:</p>
<blockquote>
<div><ol class="arabic simple">
<li><p>a single barcode identified per cell ID.</p></li>
<li><p>multiple barcodes per cell but the same barcode combination is found in other cells in the same samples.</p></li>
<li><p>multiple barcodes per cell but the same barcode combination is found in other cells across samples within the same experimental design (common for barcoding studies).</p></li>
</ol>
</div></blockquote>
</section>
<span id="document-mainSingletCode/benchmarking"></span><section id="creating-benchmark-datasets-from-singlet-code-results">
<h3>Creating Benchmark datasets from Singlet Code results<a class="headerlink" href="#creating-benchmark-datasets-from-singlet-code-results" title="Permalink to this heading">¶</a></h3>
<p>Typical benchmarking for different doublet detection methods are created by simulating doublets and using the methods to identify these cells when mixed with singlet cells in varying percentages. However, until now, there really was no ground truth for singlets in experimental data.</p>
<p>Singlet Code provides singlets that are closest possible to ground truth and can be used to simulate doublets for the benchmarking datasets.</p>
<p>To simulate the doublets, we randomly selected the count data from two cells we identify as true singlets. We then averaged the counts from these two cells to generate simulated doublets as performed previously <span id="id1">[<a class="reference internal" href="index.html#id3" title="Nicholas J. Bernstein, Nicole L. Fong, Irene Lam, Margaret A. Roy, David G. Hendrickson, and David R. Kelley. Solo: doublet identification in single-cell rna-seq via semi-supervised deep learning. Cell Systems, 11(1):95-101.e5, July 2020. doi:10.1016/j.cels.2020.05.010.">B1</a>]</span>. The final scRNA-seq datasets are generated by adding such simulated doublets into the datasets at different percentages (5-25%).</p>
</section>
</div>
</section>
<span id="document-classifier"></span><section id="classifier">
<h2>Classifier<a class="headerlink" href="#classifier" title="Permalink to this heading">¶</a></h2>
<p>This page will contain a brief introduction to the classifier and links to 4 pages: how does it work with the figure, maybe explain what is AUPRC, how does it compare to other methods, how can people integrate this into their work</p>
</section>
<span id="document-doubletDetectionMethods"></span><section id="doublet-detection-methods">
<h2>Doublet detection methods<a class="headerlink" href="#doublet-detection-methods" title="Permalink to this heading">¶</a></h2>
<p>This section will contain information about the doublet detection methods, maybe their main figure along with the link to the main paper.</p>
<p>Using the dataset we created using the results from Singlet code as described in <span class="xref std std-ref">mainSingletCode/benchmarking</span>, we benchmark four doublet detection methods - scDblFinder, DoubletFinder, Scrublet and Hybrid.</p>
<p>In this section, you can find more information about each of these methods and how they perform on the benchmark.</p>
<div class="toctree-wrapper compound">
<span id="document-otherDoubletDetectionMethod/otherMethods"></span><section id="doublet-detection-methods">
<h3>Doublet Detection Methods<a class="headerlink" href="#doublet-detection-methods" title="Permalink to this heading">¶</a></h3>
</section>
<span id="document-otherDoubletDetectionMethod/benchmark"></span><section id="benchmark">
<h3>Benchmark<a class="headerlink" href="#benchmark" title="Permalink to this heading">¶</a></h3>
</section>
<span id="document-otherDoubletDetectionMethod/AUPRCvsAUROC"></span><section id="auprc-vs-auroc">
<h3>AUPRC vs AUROC<a class="headerlink" href="#auprc-vs-auroc" title="Permalink to this heading">¶</a></h3>
</section>
</div>
</section>
<span id="document-datasetsInfo"></span><section id="datasets-used-in-singlet-code">
<h2>Datasets used in Singlet Code<a class="headerlink" href="#datasets-used-in-singlet-code" title="Permalink to this heading">¶</a></h2>
<p>This section will contain all the information about the different datasets in different tabs: the big table containin gall the information about the datasets, maybe one tab with cell types in the datasets</p>
<div class="toctree-wrapper compound">
<span id="document-DatasetsUsed/data"></span><section id="datasets-used">
<h3>Datasets used<a class="headerlink" href="#datasets-used" title="Permalink to this heading">¶</a></h3>
<p>SingletCode was tested on datasets with many different barcoding techniques, cell types and sequencing methods. Comprehensive details about the datasets used are described in the table.</p>
</section>
</div>
</section>
<span id="document-references"></span><section id="references">
<h2>References<a class="headerlink" href="#references" title="Permalink to this heading">¶</a></h2>
<p>References in the website</p>
<div class="docutils container" id="id1">
<dl class="citation">
<dt class="label" id="id3"><span class="brackets">B1</span></dt>
<dd><p>Nicholas J. Bernstein, Nicole L. Fong, Irene Lam, Margaret A. Roy, David G. Hendrickson, and David R. Kelley. Solo: doublet identification in single-cell rna-seq via semi-supervised deep learning. <em>Cell Systems</em>, 11(1):95–101.e5, July 2020. <a class="reference external" href="https://doi.org/10.1016/j.cels.2020.05.010">doi:10.1016/j.cels.2020.05.010</a>.</p>
</dd>
<dt class="label" id="id5"><span class="brackets">B2</span></dt>
<dd><p>Malte D. Luecken and Fabian J. Theis. Current best practices in single-cell rna-seq analysis: a tutorial. <em>Molecular Systems Biology</em>, 15(6):e8746, June 2019. <a class="reference external" href="https://doi.org/10.15252/msb.20188746">doi:10.15252/msb.20188746</a>.</p>
</dd>
<dt class="label" id="id4"><span class="brackets">B3</span></dt>
<dd><p>Nan Miles Xi and Jingyi Jessica Li. Benchmarking computational doublet-detection methods for single-cell rna sequencing data. <em>Cell Systems</em>, 12(2):176–194.e6, February 2021. <a class="reference external" href="https://doi.org/10.1016/j.cels.2020.11.008">doi:10.1016/j.cels.2020.11.008</a>.</p>
</dd>
<dt class="label" id="id2"><span class="brackets">B4</span></dt>
<dd><p>Ziyang Zhang, Madeline E. Melzer, Karun Kiani, and Yogesh Goyal. Singletcode: synthetic barcodes identify singlets in scrna-seq datasets and evaluate doublet algorithms. <em>bioRxiv</em>, 2023. URL: <a class="reference external" href="https://www.biorxiv.org/content/early/2023/08/07/2023.08.04.552078">https://www.biorxiv.org/content/early/2023/08/07/2023.08.04.552078</a>, <a class="reference external" href="https://doi.org/10.1101/2023.08.04.552078">doi:10.1101/2023.08.04.552078</a>.</p>
</dd>
</dl>
</div>
<p>Also, this website was built with <a class="reference external" href="https://www.sphinx-doc.org/">Sphinx</a> and hosted on <a class="reference external" href="https://readthedocs.org/">Read the Docs</a>. We would like to also acknowledge the <a class="reference external" href="https://tomobank.readthedocs.io/en/latest/index.html">TomoBank</a> as the template for building this site.</p>
</section>
<span id="document-FAQs"></span><section id="faqs">
<h2>FAQs<a class="headerlink" href="#faqs" title="Permalink to this heading">¶</a></h2>
<p>This page will contain FAQs about the paper, the classifier and about whom to contact.
Example:</p>
<blockquote>
<div><ol class="arabic simple">
<li><p>What to do if we do not use barcoded cells in our lab?</p></li>
<li><p>Where can I access the code in the paper?</p></li>
<li><p>Does the singlet detection using barcode depend on cell type?</p></li>
</ol>
</div></blockquote>
</section>
</div>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, Goyal Lab.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>